### PR TITLE
Add ability to paginate results.

### PR DESCRIPTION
**Pull request by maruqu**
This change is [Reviewable](https://reviewable.io/reviews/socialwifi/flask-jsonapi/23)

### CODE REVIEW

**Comment by mimo00** [comment]
LGTM. But we rely need "links" to use it.
JSON API doc:
The following keys MUST be used for pagination links:

    first: the first page of data
    last: the last page of data
    prev: the previous page of data
    next: the next page of data

File: flask_jsonapi/nested/nested_resource_repositories.py [hunk 1]
```diff
@@ -6,8 +6,8 @@
 
 class NestedResourceRepositoryListView(resource_repository_views.ResourceRepositoryViewMixin,
                                        flask_jsonapi.nested.nested_resources.NestedResourceList):
-    def read_many(self, filters):
-        return self.repository.get_list(filters)
+    def read_many(self, filters, pagination):
+        return self.repository.get_list(filters, pagination)
```
Comment (mimo00) [inline flask_jsonapi/nested/nested_resource_repositories.py:11]: def get_count(self, filters):
        return self.repository.get_count(filters)

File: flask_jsonapi/resource_repository_views.py [hunk 2]
```diff
@@ -26,7 +26,7 @@ def update(self, id, data, **kwargs):
 
 
 class ResourceRepositoryListView(ResourceRepositoryViewMixin, resources.ResourceList):
-    def read_many(self, filters):
+    def read_many(self, filters, pagination):
         return self.repository.get_list(filters)
```
Comment (mimo00) [inline flask_jsonapi/resource_repository_views.py]: def get_count(self, filters):
        return self.repository.get_count(filters)

File: flask_jsonapi/resource_repository_views.py [hunk 3]
```diff
@@ -26,7 +26,7 @@ def update(self, id, data, **kwargs):
...
```
Comment (mimo00) [inline flask_jsonapi/resource_repository_views.py]: Add pagination parameter to get_list

File: flask_jsonapi/query_string.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        last_page = math.ceil(total_count / page_size)
+        previous_page = current_page - 1 if current_page > 1 else None
+        next_page = current_page + 1 if current_page < last_page else None
+        return self._format_links(current_page, previous_page, next_page, last_page)
+
+    def _format_links(self, current_page, previous_page, next_page, last_page):
+        request_args = flask.request.args.copy()
+        request_args.pop('page[number]')
+        base_link = '{}?{}'.format(flask.request.base_url, parse.unquote(parse.urlencode(request_args)))
+        format_query_string = base_link + '&page=[number]={}'
```
Comment (mimo00) [inline flask_jsonapi/query_string.py]: '&page=[number]={}' should be '&page[number]={}' so without one '='

File: flask_jsonapi/nested/nested_resource_repositories.py (hunk 1 above)
Comment (maruqu) [reply flask_jsonapi/nested/nested_resource_repositories.py:11]: Added in last commit

File: flask_jsonapi/resource_repository_views.py (hunk 2 above)
Comment (maruqu) [reply flask_jsonapi/resource_repository_views.py]: added in last commit

**Comment by maruqu** [comment]
Fixed.

File: flask_jsonapi/resources.py [hunk 5]
```diff
@@ -1,9 +1,9 @@
 import http
+import logging
 
 import flask
 import marshmallow
 from flask import helpers
-from flask import logging
```
Comment (maruqu) [inline flask_jsonapi/resources.py:6]: missing in new version of flask

**Comment by jgoclawski** [comment]
LGTM.

**Pull request merged**